lib/ansible/modules/remote_management/dellemc/idrac/dellemc_idrac_server_config_profile.py

```diff
@@ -0,0 +1,312 @@
+    scp_file:   "scp_filename.xml"
+    scp_components: "ALL"
+    job_wait: True
+
+- name: Import Server Configuration Profile from a local path
+  dellemc_idrac_server_config_profile:
+    idrac_ip: "192.168.0.1"
+    idrac_user: "user_name"
+    idrac_pwd: "user_pwd"
+    command: "import"
+    share_name: "/scp_folder"
+    share_user: "share_user_name"
+    share_pwd: "share_user_pwd"
+    scp_file:   "scp_filename.xml"
+    scp_components: "ALL"
+    job_wait: True
+
+- name: Export Server Configuration Profile to a network share
+  dellemc_idrac_server_config_profile:
+    idrac_ip: "192.168.0.1"
+    idrac_user: "user_name"
+    idrac_pwd: "user_pwd"
+    share_name: "192.168.0.2:/share"
+    share_user: "share_user_name"
+    share_pwd: "share_user_pwd"
+    job_wait: False
+
+- name: Export Server Configuration Profile to a local path
+  dellemc_idrac_server_config_profile:
+    idrac_ip: "192.168.0.1"
+    idrac_user: "user_name"
+    idrac_pwd: "user_pwd"
+    share_name: "/scp_folder"
+    share_user: "share_user_name"
+    share_pwd: "share_user_pwd"
+    job_wait: False
+'''
+
+RETURN = r'''
+---
+msg:
+  type: str
+  description: Status of the import or export SCP job.
+  returned: always
+  sample: "Successfully imported the Server Configuration Profile"
+scp_status:
+  type: dict
+  description: SCP operation job and progress details from the iDRAC.
+  returned: success
+  sample:
+    {
+      "Id": "JID_XXXXXXXXX",
+      "JobState": "Completed",
+      "JobType": "ImportConfiguration",
+      "Message": "Successfully imported and applied Server Configuration Profile.",
+      "MessageArgs": [],
+      "MessageId": "XXX123",
+      "Name": "Import Configuration",
+      "PercentComplete": 100,
+      "StartTime": "TIME_NOW",
+      "Status": "Success",
+      "TargetSettingsURI": null,
+      "retval": true
+    }
+'''
+
+import os
+from ansible.module_utils.remote_management.dellemc.dellemc_idrac import iDRACConnection
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from omsdk.sdkfile import file_share_manager
+    from omsdk.sdkcreds import UserCredentials
+    from omdrivers.enums.iDRAC.iDRACEnums import (SCPTargetEnum, EndHostPowerStateEnum,
+                                                  ShutdownTypeEnum, ExportFormatEnum, ExportUseEnum)
+except ImportError:
+    pass
+
+
+def run_import_server_config_profile(idrac, module):
+    """Import Server Configuration Profile from a network share."""
+    share_name = module.params['share_name']
+    target = SCPTargetEnum[module.params['scp_components']]
+    job_wait = module.params['job_wait']
+
+    end_host_power_state = EndHostPowerStateEnum[module.params['end_host_power_state']]
+    shutdown_type = ShutdownTypeEnum[module.params['shutdown_type']]
+
+    idrac.use_redfish = True
+
+    try:
+        myshare = file_share_manager.create_share_obj(
+            share_path="{0}{1}{2}".format(share_name, os.sep, module.params['scp_file']),
```

Comment:
do you really need this `share_name` variable, I can see you are using it only one place. You can use module option directly `module.params['share_name']` right?